operator/duties/dutystore/sync_committee.go

```diff
@@ -51,20 +51,13 @@ func (d *SyncCommitteeDuties) Duty(period uint64, validatorIndex phase0.Validato
 		return nil
 	}
 
-	return descriptor.duty
+	return descriptor.Duty
 }
 
-func (d *SyncCommitteeDuties) Add(period uint64, validatorIndex phase0.ValidatorIndex, duty *eth2apiv1.SyncCommitteeDuty, inCommittee bool) {
-	d.mu.Lock()
-	defer d.mu.Unlock()
-
-	if _, ok := d.m[period]; !ok {
-		d.m[period] = make(map[phase0.ValidatorIndex]dutyDescriptor[eth2apiv1.SyncCommitteeDuty])
-	}
-
-	d.m[period][validatorIndex] = dutyDescriptor[eth2apiv1.SyncCommitteeDuty]{
-		duty:        duty,
-		inCommittee: inCommittee,
+func (d *SyncCommitteeDuties) Set(period uint64, duties []DutyDescriptor[eth2apiv1.SyncCommitteeDuty]) {
```

Comment:
you removed the mutex lock here

Reply:
@y0sher good catch, fixed
i also pushed a bunch of refactors